libs/SerialNumberKit/include/SerialNumberKit.h

```diff
@@ -5,21 +5,28 @@
 #pragma once
 
 #include <array>
+#include <filesystem>
 #include <interface/drivers/MCU.h>
+#include <span>
 
 namespace leka {
 
 class SerialNumberKit
 {
-	static constexpr size_t SN_SIZE		  = 33;
-	static constexpr size_t SHORT_SN_SIZE = 16 + 1;
+	static constexpr size_t SN_SIZE			= 31 + 1;
+	static constexpr size_t SHORT_SN_SIZE	= 15 + 1;
+	static constexpr auto date_of_test_path = "fs/sys/date_of_test";
 
   public:
 	explicit SerialNumberKit(interface::MCU &mcu) : _mcu(mcu) {}
 
 	auto getSerialNumber() -> std::array<uint8_t, SN_SIZE> &;
 	auto getShortSerialNumber() -> std::array<uint8_t, SHORT_SN_SIZE> &;
 
+	[[nodiscard]] auto setPrefix(std::span<uint8_t> serial_number) const -> uint32_t;
+	[[nodiscard]] auto setDateOfTest(std::span<uint8_t> serial_number, uint32_t offset) const -> uint32_t;
+	void setMCUID(std::span<uint8_t> serial_number, uint32_t offset, uint8_t number_of_digits);
+
```

Comment:
These methods are set `public` for unit test

And yes, they can return `std::size_t`

However, I don't get why in your suggested change you replaced `uint32_t` by `[uint32_t](std::size_t)`

Reply:
> However, I don't get why in your suggested change you replaced uint32_t by [uint32_t](std::size_t)

this was a typo, I fixed it.

> These methods are set public for unit test

is this really needed? Can't you just check the final output? having these 3 functions or just one big, or 12 smaller ones is just an implementation detail, no? 

I'd prefer to keep them private as they are never used by the user.

Reply:
> is this really needed? Can't you just check the final output? having these 3 functions or just one big, or 12 smaller ones is just an implementation detail, no?
> 
> I'd prefer to keep them private as they are never used by the user.

Not sure of this point, the other librairies do the same and only for sake of unit test. I prefer to test each step of the process rather than get a cryptic result that **might** works

Reply:
Ok for `SerialNumberFormatter`